packages/common/core-interfaces/src/brandedType.ts

```diff
@@ -7,14 +7,55 @@
  * Base branded type which can be used to annotate other type.
  *
  * @remarks
- * To use derive another class declaration and ideally add additional private
- * properties to further distinguish the type.
+ * To use, derive another class declaration and ideally add additional private
+ * and/or protected (generally preferred) properties to distinguish the type.
+ * (Any private properties should be typed as `unknown` as external to module
+ * the only thing known about a private property is that it exists.)
+ * A private constructor is recommended to seal the brand class.
+ *
+ * `BrandedType` is covariant over its type parameter, which could be leveraged
+ * for any generic type, but the preferred pattern is to specify variance
+ * explicitly in the derived class. It is convenient to have the derived class
+ * be the generic given to `BrandedType`.
  *
  * Since branded types are not real value types, they will always need to be
- * created using `as` syntax and very often `as unknown` first.
+ * created using `as` syntax and often `as unknown` first.
  *
  * This class should never exist at runtime, so it is only declared.
  *
+ * @example
+ * Definition of two branded types with different variance:
+ * ```typescript
+ * // A brand that is covariant over given T
+ * declare class CovariantBrand<T> extends BrandedType<CovariantBrand<unknown>> {
+ *    // Does not allow unrelated or less derived CovariantBrand-ed types to be
+ *    // assigned. CovariantBrand<string> is not assignable to CovariantBrand<"literal">.
+ *    protected readonly CovariantBrand: T;
+ *    private constructor();
+ * }
+ * // A brand that is contravariant over given T
+ * declare class ContravariantBrand<T> extends BrandedType<ContravariantBrand<unknown>> {
+ *    // Does not allow unrelated or more derived ContravariantBrand-ed types to be
+ *    // assigned. ContravariantBrand<"literal"> is not assignable to ContravariantBrand<string>.
+ *    protected readonly ContravariantBrand: (_: T) => void;
+ *    private constructor();
+ * }
+ * ```
+ *
+ * Applying a brand to a type through type-guard:
+ * ```typescript
+ * function numberIs5(n: number): n is number & CovariantBrand<5> {
+ *    return n === 5;
+ * }
+ * function onlyAccept4_5_or_6(_n: number & CovariantBrand<4 | 5 | 6>): void {}
+ *
+ * function example(n: number) {
+ *    if (numberIs5(n)) {
+ *       onlyAccept4_5_or_6(n); // OK: CovariantBrand<5> is assignable to CovariantBrand<4 | 5 | 6>;
+ *   }
+ * }
+ * ```
+ *
  * @sealed
```

Comment:
Given the usage pattern for this is to subclass it, I'm not sure `@sealed` makes sense. Typcially we interpret sealed to mean a type or interface which only internal FF code is allowed to implement (at least thats what our API docs generate for it). With this being internal, that doesn't really apply, but the regular meaning of no one can subclass does not make sense.

Generally in my opinion as casting something to a sealed type to get a instance of it is supposed to be invalid, which all users of the branded type have to do. I think it might make more sense to omit this tag.